### PR TITLE
fix categorical for softmax regression or in general for p with more tha 1 dimension

### PR DESCRIPTION
**Pull request by aloctavodia**
This is related to #1004

### CODE REVIEW

File: pymc3/distributions/discrete.py [hunk 1]
```diff
@@ -359,10 +359,7 @@ class Categorical(Discrete):
     """
     def __init__(self, p, *args, **kwargs):
         super(Categorical, self).__init__(*args, **kwargs)
-        try:
-            self.k = np.shape(p)[-1].tag.test_value
-        except AttributeError:
-            self.k = np.shape(p)[-1]
+        self.k = np.shape(p)[-1]
```
Comment (twiecki) [inline pymc3/distributions/discrete.py]: I think this was a recent fix by @fonnesbeck.

File: pymc3/distributions/discrete.py [hunk 2]
```diff
@@ -378,8 +375,12 @@ def logp(self, value):
         p = self.p
         k = self.k
 
-        sumto1 = theano.gradient.zero_grad(T.le(abs(T.sum(p) - 1), 1e-5))
-        return bound(T.log(p[value]),
+        sumto1 = theano.gradient.zero_grad(T.le(abs(T.sum(p, axis=-1) - 1), 1e-5))
+        if p.ndim > 1:
+            a = T.log(p[T.arange(p.shape[0]), value])
```
Comment (twiecki) [inline pymc3/distributions/discrete.py:384]: I think we really need a test for both of these cases.

File: pymc3/distributions/discrete.py (hunk 1 above)
Comment (fonnesbeck) [inline pymc3/distributions/discrete.py]: Yes, we need to leave this in, otherwise `k` ends up being a tensor. It needs to be a scalar.

**Comment by twiecki** [comment]
Thanks @aloctavodia. If you could add a test later on, that'd be great.

**Pull request merged**